ui-design/virtual-machines/clone-vm/clone-vm.md

```diff
@@ -0,0 +1,33 @@
+# Clone VM
+
+![Offline VM in List View](img/2-0-vms.png)
+
+Cloning a VM allows the user to quickly create an identical copy of a virtual machine while powered off.
+
+![VM List View Clone option in kebab menu](img/2-1-vm-list.png)
+
+Cloning is accessed from the VM List View. All item filters should be turned on by default, ensuring that all VMs are shown to the user.
+
+The Clone option should appear in the kebab menu whether the target is online or offline.
+
+![VM modal with still running notice](img/2-2-modal.png)
+
+If the VM is powered on, a warning will appear at the top of the Clone Virtual Machine modal. Running VMs will be shut down automatically if the user proceeds with a clone and then turned back on once the clone finishes.
+
+The only required field is the name, which is pre-filled based on the target VM’s name and appended with “-clone”. The user can add a description and change the new VM’s Namespace (which is the same as the target by default). They can also turn off the option to start the virtual machine after the cloning process finishes.
+
+Because all required fields are filled in by default, the "Clone Virtual Machine” button is active. If the user deletes the Name field this button will become inactive.
+
+Details of the target VM’s configuration are displayed for the user’s reference. These options cannot be modified here prior to cloning.
+
+The user would then click “Clone Virtual Machine”.
+
+![Clone VM power off failure](img/2-3-vm-list-error.png)
+
+If the target VM fails to shut down properly, an error toast notification would be displayed. The user would then need to power off the VM manually and re-open the Clone modal to try again.
+
+![Clone VM started](img/2-4-vm-list-cloning.png)
+
+If the cloning process starts successfully, the newly-cloned VM will appear in the list and a toast notification will confirm that the cloning process has begun. The status icon of the clone is a spinner with a completion percentage next to it. The clone does not have an IP address until after its first boot.
+
```

Comment:
We should disable certain operations on the source VM during a clone operation (such as starting the VM).  Perhaps the VM's status icon should change to a lock or something?  We will need to figure out a way to lock VMs from starting (maybe an addition to the VM CRD?)

Reply:
Good point - I've changed the status icon to a lock and added two screens showing which options should be available while cloning. How do these look?

I'm not familiar with the technical implementation details, but I'm guessing that the source VM's configuration could still be changed since the clone's parameters have likely already been specified. These screens also assume that two clones can be made at the same time from the same source.